### PR TITLE
Use GovUKNotify as email sender

### PR DESCRIPTION
**Pull request by colinbruce**
## Why?
Issues with previous provider caused issues with third party AV/firewalls

## How?
Switching to GovUK Notify will save money and allow simpler email patterns

### CODE REVIEW

File: app/controllers/concerns/password_helpers.rb [hunk 1]
```diff
@@ -19,7 +19,14 @@ def update_password
...
   def deliver_reset_password_instructions(user)
-    user.send_reset_password_instructions
+    token, enc = Devise.token_generator.generate(user.class, :reset_password_token)
+    user.reset_password_token   = enc
+    user.reset_password_sent_at = Time.now.utc
+    user.save(validate: false)
+    message = DeviseMailer.reset_password_instructions(user, token, current_user.name)
+    message.deliver_now
+  rescue => e
+    Rails.logger.error("DEVISE MAILER ERROR: '#{e.message}' while sending reset password mail")
```
Comment (jsugarman) [inline app/controllers/concerns/password_helpers.rb]: should we raise the exception to allow sentry to pick it up, or are our logs being alerted on

File: config/environments/development.rb [hunk 2]
```diff
@@ -73,8 +73,8 @@
 
 
   # normal dev mail configuration
-  config.action_mailer.perform_deliveries = false
-  config.action_mailer.delivery_method = :file
+  config.action_mailer.perform_deliveries = Settings.govuk_notify.api_key.present?
+  config.action_mailer.raise_delivery_errors = false
```
Comment (jsugarman) [inline config/environments/development.rb]: a discussed, for development we probably want to see the error

File: config/initializers/config.rb [hunk 3]
```diff
@@ -1,31 +1,15 @@
 Config.setup do |config|
   # Name of the constant exposing loaded settings
   config.const_name = 'Settings'
+  config.use_env = true
+  config.env_prefix = 'SETTINGS'
+  config.env_separator = '__'
```
Comment (jsugarman) [inline config/initializers/config.rb:6]: is the double underscore deliberate?

File: spec/mailers/previews/notify_mailer_preview.rb [hunk 4]
```diff
@@ -1,8 +1,8 @@
 class NotifyMailerPreview < ActionMailer::Preview
```
Comment (jsugarman) [inline spec/mailers/previews/notify_mailer_preview.rb:2]: should this whole file not be removed?

**Review by jsugarman — APPROVED** [review]
Aside from some little comments/queries/double-checks. it looks good

Also you can probably amend the `Gemfile` to remove, at least, `gem 'premailer-rails',         '~> 1.9'`

File: spec/mailers/previews/notify_mailer_preview.rb (hunk 4 above)
Comment (jsugarman) [reply spec/mailers/previews/notify_mailer_preview.rb:2]: ..and 
- app/assets/stylesheets/email.css
- app/views/layouts/email.html.haml

File: config/initializers/config.rb (hunk 3 above)
Comment (colinbruce) [reply config/initializers/config.rb:6]: Yes, it's to allow users to use underscores in env vars (if they need to?!) while allowing config to split a double underscore into dotted strings e.g. test__value becomes `Settings.test.value`, while test_value would remain `Settings.test_value`

**Pull request merged**